### PR TITLE
Add support for users and fetching private messages

### PR DESCRIPTION
**Pull request by mstump**
I added support for fetching user information and fetching private messages. It required expanding the messages API which required changing the signature of some of the types.

### CODE REVIEW

File: tests/lib.rs [hunk 1]
```diff
@@ -11,7 +11,8 @@ use hipchat_client::Client as HipchatClient;
 struct Config {
     token: String,
     origin: String,
-    room: String
+    room: String,
+    user: String
```
Comment (rsolomo) [inline tests/lib.rs]: When trying to run the integration tests, I initially found myself using my username for the configuration, which would not work.

To make the intention more clear, could we update this type to be a number, and update the example.settings.json?

Comment (mstump) [inline tests/lib.rs]: Yes. I'll make the change and update the pull request.

**Comment by mstump** [comment]
Changes made, checks passed

**Comment by rsolomo** [comment]
Thanks! :smile:

**Pull request merged**